### DISCUSSION TITLE
Exact match in `FilterByInput`

### DISCUSSION THREAD

**Discussion by loxy** [taiga-family/taiga-ui#2373] [Q&A] [answered]
Hi!

When we use  `tuiFilterByInput` pipe and insert a search string which is an exact match, the filtered items are gone and it shows the whole list. You can see it on the [website](https://taiga-ui.dev/pipes/filter-by-input) in first (Basic) example. Just type in "John Cleese".

Is this by intention? If yes: why? This leads to UX problems if you have a long list and user is not aware that he has already selected the right one...

Best regards,
Kersten

**Comment by waterplea** [comment, marked as answer]
Yes, it is by intention, otherwise in ComboBox when some value is selected — you will never see other options, unless you clear it first. I believe it's supposed to scroll you to the matched value in the list, showing the checkmark next to it. If it doesn't — it's a bug.

**Reply by loxy** [reply]
Ah, ok. Thanks! Yes, this works, I just tried it. But we are using it together with a MultiSelect. Is it only for usage in ComboBox?

**Reply by waterplea** [reply]
It's for any control with text input and a dropdown. Shouldn't user see the selected one in the control itself? Anyway, it's pretty hard make a universally well solution here, current one seems to be good enough. If you have practical idea on how to improve it, please share them. Otherwise you can take a look at the source code and copy it into your project, altering to suit your needs.
